staff/views/redirects.py

```diff
@@ -48,15 +50,26 @@ def get_queryset(self):
         qs = super().get_queryset()
 
         if q := self.request.GET.get("q"):
-            qs = qs.filter(
-                Q(old_url__icontains=q)
-                | Q(project__name__icontains=q)
-                | Q(workspace__name__icontains=q)
-                | Q(created_by__username__icontains=q)
-                | Q(created_by__fullname__icontains=q)
-            ).distinct()
+            fields = [
+                "analysis_request__title",
+                "created_by__fullname",
+                "created_by__username",
+                "old_url",
+                "org__name",
+                "project__name",
+                "workspace__name",
+            ]
+
+            # build up Q objects OR'd together.  We need to build them with
+            # functools.reduce so we can optionally add the PK filter to the
+            # list
+            qwargs = functools.reduce(
+                Q.__or__, (Q(**{f"{f}__icontains": q}) for f in fields)
```

Comment:
I get this, but also :scream:

Slight suggestion to improve future readability
```suggestion
                Q.__or__, (Q(**{f"{field}__icontains": q}) for field in fields)
```

Reply:
this is amazing